UnitTests/BraintreePayPalNativeCheckoutTests/BTPayPalNativeCheckoutClient_Tests.swift

```diff
@@ -28,6 +40,176 @@ class BTPayPalNativeCheckoutClient_Tests: XCTestCase {
+            "paypalEnabled": true,
+            "environment": "sandbox",
+            "paypal": ["clientId": "a-fake-client-id"] as [String: Any?]
+        ] as [String: Any])
+
+        apiClient.cannedResponseBody = BTJSON(value: [
+            "paymentResource": ["redirectUrl": "https://www.paypal.com/checkout?token=fake-ec-token"]
+        ])
+
+        let payPalNativeCheckoutClient = BTPayPalNativeCheckoutClient(apiClient: apiClient, nativeCheckoutProvider: mockNativeCheckoutProvider)
+        let request = BTPayPalNativeCheckoutRequest(amount: "1.99")
+        payPalNativeCheckoutClient.tokenize(request) { _, _ in }
+        mockNativeCheckoutProvider.triggerCancel()
+        XCTAssertTrue(mockNativeCheckoutProvider.didCancel)
+    }
+
+    func testTokenize_whenOnStartableErrorCalled_returnsError() {
+        apiClient.cannedConfigurationResponseBody = BTJSON(value: [
+            "paypalEnabled": true,
+            "environment": "sandbox",
+            "paypal": ["clientId": "a-fake-client-id"] as [String: Any?]
+        ] as [String: Any])
+
+        let payPalNativeCheckoutClient = BTPayPalNativeCheckoutClient(apiClient: apiClient, nativeCheckoutProvider: mockNativeCheckoutProvider)
+        let request = BTPayPalNativeCheckoutRequest(amount: "1.99")
+        payPalNativeCheckoutClient.tokenize(request) { _, error in
+            self.mockNativeCheckoutProvider.triggerError(error: error as! BTPayPalNativeCheckoutError)
+            XCTAssertEqual(error as! BTPayPalNativeCheckoutError, BTPayPalNativeCheckoutError.orderCreationFailed(BTPayPalNativeCheckoutError.invalidJSONResponse))
+        }
+    }
+
+    func testTokenize_whenInvalidRedirectURL_returnsError() {
+        apiClient.cannedConfigurationResponseBody = BTJSON(value: [
+            "paypalEnabled": true,
+            "environment": "sandbox",
+            "paypal": ["clientId": "a-fake-client-id"] as [String: Any?]
+        ] as [String: Any])
+
+        apiClient.cannedResponseBody = BTJSON(value: [
+            "paymentResource": ["redirectUrl": "not-a-url"]
+        ])
+
+        let payPalNativeCheckoutClient = BTPayPalNativeCheckoutClient(apiClient: apiClient)
+        let request = BTPayPalNativeCheckoutRequest(amount: "1.99")
+        let expectation = expectation(description: "Checkout fails with error")
+
+        payPalNativeCheckoutClient.tokenize(request) { nonce, error in
+            guard let error = error as NSError? else { XCTFail(); return }
+            XCTAssertNil(nonce)
+            XCTAssertEqual(error.domain, BTPayPalNativeCheckoutError.errorDomain)
+            XCTAssertEqual(error.code, BTPayPalNativeCheckoutError.orderCreationFailed(BTPayPalNativeCheckoutError.invalidJSONResponse).errorCode)
+            XCTAssertEqual(error.localizedDescription, BTPayPalNativeCheckoutError.orderCreationFailed(BTPayPalNativeCheckoutError.invalidJSONResponse).errorDescription)
+            expectation.fulfill()
```

Comment:
Nitpick - I think ideally the tests should assert on the raw string values & raw int codes expected, versus referring to constants in the SDK. Otherwise we don't catch spelling errors or typos.